pool/pool_manager.gno

```diff
@@ -30,6 +51,87 @@ func init() {
 	feeAmountTickSpacing[10000] = 200 // 1%
 }
 
+type poolInput struct {
+	token0Path, token1Path string
+	fee                    uint32
+	sqrtPriceX96           *u256.Uint
+	tickSpacing            int32
+}
+
+func newPoolInput(
+	token0Path string,
+	token1Path string,
+	fee uint32,
+	sqrtPriceX96 string,
+) *poolInput {
+	price := u256.MustFromDecimal(sqrtPriceX96)
+	tickSpacing := feeAmountTickSpacing.Get(fee)
+	return &poolInput{
+		token0Path: token0Path,
+		token1Path: token1Path,
+		fee:        fee,
+		sqrtPriceX96: price,
+		tickSpacing: tickSpacing,
+	}
+}
+
+func (p *poolInput) isSameTokenPath() bool {
+	return p.token0Path == p.token1Path
+}
+
+func (p *poolInput) isInOrder() bool {
+	if strings.Compare(p.token0Path, p.token1Path) > 0 {
+		return false
+	}
+
+	return true
+}
+
+func (p *poolInput) wrap() (string, string) {
+	if p.token0Path == consts.GNOT {
+		p.token0Path = consts.WRAPPED_WUGNOT
+	}
+	if p.token1Path == consts.GNOT {
+		p.token1Path = consts.WRAPPED_WUGNOT
+	}
+
+	return p.token0Path, p.token1Path
+}
+
+func (p *poolInput) filterValidTokenPaths() (string, string, error) {
+	if p.isSameTokenPath() {
+		return "", "", errDuplicateTokenInPool
+	}
+
+	token0Path, token1Path := p.wrap()
+
+	if !p.isInOrder() {
+		return "", "", errTokenSortOrder
+	}
+
+	return token0Path, token1Path, nil
+}
```

Comment:
seems like `filterValidTokenPaths()` will pass if 
 - token0 = ugnot
 - token1 = wunogt 
or vice versa.

you also need to check if tokens are same after wrapping